.github/workflows/cla-assistant.yaml

```diff
@@ -0,0 +1,47 @@
+name: "CLA Assistant"
+on:
+  issue_comment:
+    types: [created]
+  pull_request_target:
+    types: [opened, closed, synchronize]
+# explicitly configure permissions, in case your GITHUB_TOKEN workflow permissions are set to read-only in repository settings
+permissions:
+  actions: read
+  contents: read # this can be 'read' if the signatures are in remote repository
+  pull-requests: write
+jobs:
+  CLAAssistant:
+    runs-on: ubuntu-latest
+    steps:
+      - id: fetch-token
+        name: Fetch installation token
+        uses: actions/create-github-app-token@v1
+        with:
+          app-id: ${{ secrets.CLA_ASSISTANT_APP_ID }}
+          private-key: ${{ secrets.CLA_ASSISTANT_APP_PRIVATE_KEY }}
+          repositories: cla-signatures
+      - id: get-membership
+        name: "Determine gravitational membership"
+        env:
+          GITHUB_TOKEN: ${{ steps.fetch-token.outputs.token }}
+          USER: ${{ github.actor }}
+        continue-on-error: true
+        # Use the "Check Membership" API for user
+        # If the user is a member of the org expect a 204 HTTP Status Code
+        # If the above is success the process will exit 0
+        run: |
+          gh api "orgs/gravitational/members/${USER}"
+      - name: "CLA Assistant"
+        if: steps.get-membership.outcome != 'success' && ((github.event.comment.body == 'recheck' || github.event.comment.body == 'I have read the CLA Document and I hereby sign the CLA') || github.event_name == 'pull_request_target')
```

Comment:
I'm not sure I fully understand this condition. Looks like it checks if the comment says either "recheck" or "I have read ..." and adds them to the signers repo? Does it check that the comment was left by the PR author?

Reply:
You're right it is a bit vague. I'll add some comments to describe the expected behavior.

Essentially though the behavior for anyone outside gravitational org is:
* CLA Assistant gathers authors from commits made to the PR
* If any author has not signed CLA it fails.

In the case of a failure the DX is:
* Reply with "I have read..."
  * The action will add the username to the cla-signatures list and "recheck" automatically
* "recheck" (can be commented to manually rerun in the case of a transient error)
  * Gathers authors from all commits in PR and checks cla-signatures